### PR TITLE
Asurascans Hide Premium chapters

### LINKED ISSUE #6614: Hide Locked Chapter in Asura
### Source name

Asura Scans

### Source language

English

### Describe your suggested feature

IMG_20241215_032018
IMG_20241215_032037
Asura has introduced a subscription model.
Some of the latest chapters are locked, they unlock in ~6 hr after release. 

So, please add a toggle option to hide those paid/locked chapters.
IMG_20241215_113129
P.S.: Login with username/password is not part of this request, there is a separate issue for that.

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/mihonapp/mihon/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/keiyoushi/extensions-source/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Creepler13**
Closes:  #6614

Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Have not changed source names
- [X] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/en/asurascans/src/eu/kanade/tachiyomi/extension/en/asurascans/AsuraScans.kt [hunk 1]
```diff
@@ -154,8 +154,11 @@ class AsuraScans : ParsedHttpSource(), ConfigurableSource {
     }
 
     private fun getGenreFilters(): List<Genre> = genresList.map { Genre(it.first, it.second) }
-    private fun getStatusFilters(): List<Pair<String, String>> = statusesList.map { it.first to it.second.toString() }
-    private fun getTypeFilters(): List<Pair<String, String>> = typesList.map { it.first to it.second.toString() }
+    private fun getStatusFilters(): List<Pair<String, String>> =
+        statusesList.map { it.first to it.second.toString() }
+
+    private fun getTypeFilters(): List<Pair<String, String>> =
+        typesList.map { it.first to it.second.toString() }
```
Comment (vetleledaal) [inline src/en/asurascans/src/eu/kanade/tachiyomi/extension/en/asurascans/AsuraScans.kt]: Don't change the style of unrelated code unless the linter complains. It just adds noise to the diff.

**Comment by bapeey** [comment]
preference should be enabled by default imo

**Review by AwkwardPeak7 — APPROVED** [review]
